acceptance-tests/src/test/java/it/com/atlassian/bitbucket/jenkins/internal/SmokeTest.java

```diff
@@ -442,6 +517,84 @@ private void runFullBuildFlow(BitbucketScmWorkflowJob workflowJob) throws IOExce
                 assertThat(status, successfulBuildWithKey(getBuildKey(workflowJob))));
     }
 
+    private void runFullFlowMultiBranch(boolean triggerOnRefChange, boolean triggerOnPullRequest) throws IOException, GitAPIException, InterruptedException {
+        BitbucketScmWorkflowMultiBranchJob multiBranchJob =
+                jenkins.jobs.create(BitbucketScmWorkflowMultiBranchJob.class);
+        BitbucketBranchSource bitbucketBranchSource = multiBranchJob.addBranchSource(BitbucketBranchSource.class);
+        bitbucketBranchSource
+                .credentialsId(bbsAdminCredsId)
+                .serverId(serverId)
+                .projectName(forkRepo.getProject().getKey())
+                .repositoryName(forkRepo.getSlug());
+        multiBranchJob.enableBitbucketWebhookTrigger(triggerOnRefChange, triggerOnPullRequest);
+        multiBranchJob.save();
+
+        // Clone (fork) repo
+        File checkoutDir = tempFolder.newFolder(REPOSITORY_CHECKOUT_DIR_NAME);
+        Git gitRepo = cloneRepo(ADMIN_CREDENTIALS_PROVIDER, checkoutDir, forkRepo);
+
+        // Push new Jenkinsfile to master
+        RevCommit masterCommit =
+                commitAndPushFile(gitRepo, ADMIN_CREDENTIALS_PROVIDER, MASTER_BRANCH_NAME, checkoutDir,
+                        JENKINS_FILE_NAME, ECHO_ONLY_JENKINS_FILE_CONTENT.getBytes(UTF_8));
+        String masterCommitId = masterCommit.getId().getName();
+
+        //trigger scanning so we get no surprise builds on other branches later
+        multiBranchJob.reIndex();
+        multiBranchJob.waitForBranchIndexingFinished(30);
+
+        // Push Jenkinsfile to feature branch
+        final String featureBranchName = "test-feature";
+        gitRepo.branchCreate().setName(featureBranchName).call();
+        RevCommit featureBranchCommit =
+                commitAndPushFile(gitRepo, ADMIN_CREDENTIALS_PROVIDER, featureBranchName, checkoutDir,
+                        JENKINS_FILE_NAME, ECHO_ONLY_JENKINS_FILE_CONTENT.getBytes(UTF_8));
+        String featureBranchCommitId = featureBranchCommit.getId().getName();
+
+        //wait for indexing to complete
+        multiBranchJob.waitForBranchIndexingFinished(30);
+        //this is terrible but we must give Jenkins a chance to react to the trigger and schedule a build
+        Thread.sleep(Duration.ofSeconds(10).toMillis());
+
+        Build build = multiBranchJob.getJob(featureBranchName).getLastBuild();
+        if (triggerOnRefChange && !triggerOnPullRequest) {
+            assertThat("Wrong started state of build after ref change", build.hasStarted(), is(true));
+        }
+        if (triggerOnPullRequest && !triggerOnRefChange) {
+            assertThat("Wrong started state of build after ref change", build.hasStarted(), is(false));
+        }
+        if (triggerOnRefChange && triggerOnPullRequest) {
```

Comment:
I don't think we need this 3rd if block. If `triggerOnRefChange` is true then we don't care about `triggerOnPullRequest` we just care that the build has started.